### PR TITLE
Major revision and update of the syllabus

### PR DESCRIPTION
**Pull request by lwjohnst86**
If everyone can review this one, that would be amazing :grin:

### CODE REVIEW

File: chapters/01-syllabus.Rmd [hunk 1]
```diff
@@ -1,181 +1,152 @@
...
+
+- The main package used throughout the book is the Zipf's Law package `zipfs`.
+This package is used for the code-alongs, the actual content, and most of the
+exercises.
+- The example package for the first project assignment is the Weather in Kenya
+package `kenyaweather`. This package is used as a reference when you work on
+your package for the first project assignment.
+- A currently existing package that you'll contribute to, which is the second
+project assignment.
+- (Optional) A group-based package, which is the third project assignment.
```
Comment (mbonsma) [inline chapters/01-syllabus.Rmd:137]:
```suggestion
- Optionally, you will create a package in a group for the third project assignment.
```

**Review by mbonsma — COMMENTED** [review]
Looks good! Mostly I'm thinking we should re-use as much as we can from the Python intro, but correct me if I'm misunderstanding the purpose of this document!

**Comment by lwjohnst86** [comment]
> Looks good! Mostly I'm thinking we should re-use as much as we can from the Python intro, but correct me if I'm misunderstanding the purpose of this document!

Thanks Maddy! Just want to address this specifically for now. We should definitely re-use from the Python intro, but there are and will be big differences. Also, the Python book doesn't explicitly have a syllabus, so in that regard there will be some differences.

File: chapters/01-syllabus.Rmd [hunk 2]
```diff
@@ -1,181 +1,152 @@
...
+increasingly have massive datasets or use high performance computing servers,
+along with a trend for a greater demand that scientific studies be
+reproducible and open. This book will teach you the tools, skills, and knowledge
+necessary to make research software that is open, reproducible, sustainable, and
+more rigorous. (TODO: other word?) First and foremost, the main learning goal of
+this course is that you learn to **create and develop an R package**.
+More specifically, the learning objectives are to:
+
+- Explain what an R package is, how it is structured, and how to set one up.
+- List and describe what some best (or good enough) practices are in package
```
[the remark this answers is not in the document]
Comment (k8hertweck) [reply chapters/01-syllabus.Rmd:13]: Why only some best practices? If I were a reader, I'd be wondering what I'm missing.

File: chapters/01-syllabus.Rmd [hunk 3]
```diff
@@ -1,181 +1,152 @@
...
+development and how to implement them. 
+- Explain what a code style guide is, why it is important, and how to write code
+that adheres to the style guide.
+- Describe why user-friendly and well-written documentation is crucial to
+building a usable package and some steps to take to achieve this type of
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:18]:
```suggestion
building a usable package and some steps to achieve this type of
```

File: chapters/01-syllabus.Rmd [hunk 4]
```diff
@@ -1,181 +1,152 @@
...
+documentation.
+- Cover what CRAN is, what policies and quality-control steps it has for packages,
+and how to prepare a package for submission to CRAN so that it passes these checks
+and requirements.
+
+Aside from the direct technical aspects of building an R package, the course
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:24]:
```suggestion
Aside from the technical aspects of building an R package, the course
```

File: chapters/01-syllabus.Rmd [hunk 5]
```diff
@@ -1,181 +1,152 @@
...
+will also cover topics that enhance and improve the process of package development.
+These learning objectives are to:
+
+- Describe and apply effective workflows and project management tools that help
+to coordinate the development of software, particularly for software that is
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:29]:
```suggestion
- Describe and apply effective workflows and project management tools that
coordinate the development of software, particularly for software that is
```

File: chapters/01-syllabus.Rmd [hunk 6]
```diff
@@ -1,181 +1,152 @@
...
+complex or when working in a team.
+- Outline (and optionally practice) how to work productively in a small team,
+that is inclusive and welcoming.
+- Explain the importance of and learning to design software and documentation
+that involves the community, such as by gaining potential contributors.
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:34]:
```suggestion
- Explain the importance of and learning to design software and documentation
involving community contributors.
```

File: chapters/01-syllabus.Rmd [hunk 7]
```diff
@@ -1,181 +1,152 @@
...
+
+TODO: Include objectives about having ORCID and getting DOIs?
+
+Overall, we'll be addressing the questions:
+
+- What is an R package and why should I create one?
+- How can I build an R package that is re-usable, well-built, and well-documented?
+- What is CRAN and should I submit my package to it? If so, how do I do it?
+- How can I work effectively as a team on building an R package or to
+contribute to an existing package?
```
[the remark this answers is not in the document]
Comment (k8hertweck) [reply chapters/01-syllabus.Rmd:44]: I'm also not sure I like listing this set of summary objectives. I think they should be converted from questions to statements, and added to the section below ("by the end of the course")

File: chapters/01-syllabus.Rmd [hunk 8]
```diff
@@ -1,181 +1,152 @@
...
+
+- Describe and apply effective workflows and project management tools that help
+to coordinate the development of software, particularly for software that is
+complex or when working in a team.
+- Outline (and optionally practice) how to work productively in a small team,
+that is inclusive and welcoming.
+- Explain the importance of and learning to design software and documentation
+that involves the community, such as by gaining potential contributors.
+
+TODO: Include objectives about having ORCID and getting DOIs?
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:36]: I'm not personally in favor of creating an objective for ORCID and DOIs, as that will require us to go through the process ourselves. I think it *should* be included in the narrative, but as a discussion point as another option.

File: chapters/01-syllabus.Rmd [hunk 9]
```diff
@@ -1,181 +1,152 @@
...
+- How can I build an R package that is re-usable, well-built, and well-documented?
+- What is CRAN and should I submit my package to it? If so, how do I do it?
+- How can I work effectively as a team on building an R package or to
+contribute to an existing package?
+
+By the end of the course, learners are expected to have the skills and knowledge
+necessary to build their own packages and to, if relevant, submit these packages
+to CRAN.
+
+TODO: Include bioconductor?
```
Comment (k8hertweck) [inline chapters/01-syllabus.Rmd:50]: I'm also not in favor of including bioconductor (except as a brief mention in the chapter); it involves adding too much extra detailed conceptual framework about how installation works, submissions to the repository, and why the objects are so big and weird.

**Review by k8hertweck — COMMENTED** [review]
This is super useful. I still need to look through the rest of the chapters, but can test to see whether the book content looks like it'll be delivering what the syllabus describes.

As I think about it, I'm wondering if we should rename this chapter, to help disambiguate it with the instructor's own syllabus used for university/course policies and deadlines?

File: chapters/01-syllabus.Rmd (hunk 9 above)
Comment (joelostblom) [reply chapters/01-syllabus.Rmd:50]: Yeah, will we do something specifically bio-related that requires knowing bioconductor?

**Review by joelostblom — COMMENTED** [review]
Agree with most of what was said already, but not much to add.

File: chapters/01-syllabus.Rmd [hunk 10]
```diff
@@ -1,181 +1,152 @@
...
+whether the package is well tested and easy to install: sooner or later
+(probably sooner) it will become [abandonware][abandonware].
+
+## Overview of the packages created throughout course
+
+Throughout the book, you will build several packages. There is the main
+one (`zipfs`) and 2 or 3 project assignment packages (see the assignment
+descriptions in Appendix \@ref(course-assignments)). For two of the packages,
+we have already built them so you can use them as an example and a reference
+(`zipfs` and `kenyaweather`).
```
[the remark this answers is not in the document]
Comment (lwjohnst86) [reply chapters/01-syllabus.Rmd:126]: Hmm, not sure this is the best place for it, but also not sure right now where a better place would be. Adding for now, we can decide later.

**Comment by lwjohnst86** [comment]
I've made many of the suggested changes, plus from our meeting discussion. I'm merging for now, it can be reviewed again at the next review cycle.

**Pull request merged**